discord-bot/index.js

```diff
@@ -52,12 +52,15 @@ client.on('interactionCreate', async interaction => {
 run();
 
 async function run() {
-  console.log('Connecting to', process.env.STARGATE_JSON_API_URL);
-  await mongoose.connect(process.env.STARGATE_JSON_API_URL, {
+  const url = process.env.ASTRA_JSON_API_URL || process.env.STARGATE_JSON_API_URL;
+  const isAstra = process.env.ASTRA_JSON_API_URL;
+  const options = isAstra ? { createNamespaceOnConnect: false } : {
```

Comment:
Does `createNamespaceOnConnect` throw an error with Astra, or is this change just to avoid an extra unnecessary API call?

Reply:
Trying to create a namespace via the API throws an error, so needed to pass this flag for Astra connection.

Reply:
> I just did a quick review, haven't tested yet. One overall bit of feedback is that I'm not liking how many code changes we need to make for Astra vs local Stargate.
> 
> I don't have a good suggestion for how to reconcile the two yet. But my ideal situation would be a single environment variable, so the user's code has nothing that's astra-specific or stargate-specific.

Probably, since we parse the URL, we can check if the base url has the domain, we can interpret that as Astra in the driver code and control things internally. Or we should get a flag from the User to say whether it is Astra connection string or the open source JSON API url, then we can have a common code.

Reply:
Couple of options from here:
- Option1: We can let the user pass a flag to say whether the connection string is an Astra Connection URL or a JSON API, so that we don't have to rely on the domain name or anything like that (since it might change at some point).
- Option2: We can get the individual elements as Jeff suggested and construct the URL and give preference to Astra configs when both Astra and the Stargate configurations are present.

Reply:
Maybe another option would be to assume that if `authUrl` option is set, we consider it a Stargate connection. If `authUrl` is not set, we consider it an Astra connection. What do you think?

Reply:
The problem is even with Astra we can give `authUrl`, `username` and `password` if we would like.

For example: 
```
AstraURL: https://64c2aea9-0390-4ae0-bf5e-f41ad360d340-eu-west-1.apps.astra-test.datastax.com/api/json/v1/json_store
username: xasdsadadasd
password: asdadasdada
authUrl: https://64c2aea9-0390-4ae0-bf5e-f41ad360d340-eu-west-1.apps.astra-test.datastax.com/api/rest/v1/auth
```

So the problem now we have is, we have two types of URLs with same set of options. 
The reason why we need to identify in the driver code what type it is, is because we should not be performing `createDatabase` on connection,  as part of `createCollection`, etc.

Why don't we get a simple flag from user whether it is Astra or not, while connecting (something like below)? which will simplify a lot of things on our side and user also will be clear whether to give this flag or not.

`mongoose.connect('connectionUrl', {isAstra: true})`

Reply:
That makes sense, an isAstra option is reasonable in that case

Reply:
Cool. I think, this is going to simplify multiple things that are in-flight now. Let me make this change in the driver side and submit a PR.